wire-schema/src/main/java/com/squareup/wire/schema/Options.kt

```diff
@@ -198,6 +214,33 @@ class Options(
     }
   }
 
+  private fun union(
+    linker: Linker, a: List<LinkedOptionEntry>, b: List<LinkedOptionEntry>
+  ): List<LinkedOptionEntry> {
+    val aMap: Map<ProtoMember, Any?> = a.toMap()
+    val bMap: Map<ProtoMember, Any> = b.toMap() as Map<ProtoMember, Any>
+
+    val result: MutableMap<ProtoMember, Any?> = LinkedHashMap(aMap)
+    for (entry in bMap) {
+      val bValue = entry.value
+      val aValue = result[entry.key]
+      result[entry.key] = when {
+        aValue != null -> union(linker, aValue, bValue)
+        else -> bValue
+      }
+    }
+
+    return a.map { it.optionElement to it.protoMember }
```

Comment:
this reads really awkwardly, especially as we have a.map and aMap

Reply:
I agree